samples/drivers/uart/single_wire/README.rst

```diff
@@ -0,0 +1,30 @@
+.. _single_wire:
```

Comment:
This sample seems to be board-specific, so it should be moved to `samples/boards`. Moreover, please add sample into a separate commit.

Reply:
I would rather try to get hands on some different boards and add further overlays.

Reply:
Separated the commits.

Reply:
I agree sample location should be adapted to be more specific.
This being said, having platform specific drivers samples/tests is a common thing (cf `tests/drivers/can`, `tests/drivers/clock_control`,), so I think `samples/drivers/uart/stm32/single_wire` should work.
@gmarull, what do you think about this proposal?

Reply:
As far as it's clear that the sample is specific for STM32 targets I'm fine with it.

Reply:
sounds good, will move the example

Reply:
Done